### ISSUE TITLE
FO - Combination option not added in the select

### ISSUE THREAD

**Issue by guestisp** [PrestaShop/PrestaShop#30960]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

When you have different attributes combinations, the select option in frontoffice is not "added" properly.

Let's assume you have a product configured like this:

![image](https://user-images.githubusercontent.com/3944655/213493682-3e32cd12-afd0-42a4-b601-b7650a3ac15b.png)

as you can see, for `Supporto - Nessuno` we only have `Numero di supporti - Nessuno` as combition option
every other has all combination except the `Numero di supporti - Nessuno`

This makes impossible, in the frontoffice, to choose the "Supporto: Nessuno" attribute because the select option is missing (the page loads with "Arancione, 1, Ventosa" selected, if you switch to "Nessuno", the select option is missing.

### Steps to reproduce

1. Add these attributes: "Color", "Supporto", "Numero di supporti "
2. Add these values: "Color: [red, green, blue]", "Mount type: [Nessuno, Magnetico, Ventosa]", "Numero di supporti : [Nessuno, 1, 2]"
3. Create a test product
4. Create all combinations
5. For each color AND "Nessuno" as "Supporto" attribute, remove all combination EXCEPT "Nessuno" from "Numero di supporti" attribute
6. For each color AND all values in "Supporto" attribute EXCEPT "Nessuno", remove "Nessuno" from "Numero di supporti" attribute
7. in the frontoffice you can't choose "Supporto: Nessuno" probably because the select options are not the same in all attributes and noone is adding the "Nessuno" option in the select

### PrestaShop version(s) where the bug happened

8.0.1

**Comment by florine2623** [comment]
Hello @guestisp ,

I followed your steps until step 4. I have 27 combinations and no error in FO. I can choose any combinations I want that is available in BO.

https://user-images.githubusercontent.com/16019289/214805305-15f12bcc-222e-409d-b555-e53d0fceb79a.mov

I don't understand steps 5 ands 6.

If you don't want zll of the attributes values, you don't have to add them in the first place.
For example, I added all colors and chose only the values of Mount Type and Numero di supporti that I want :


https://user-images.githubusercontent.com/16019289/214805945-ac7905ac-42ec-4ddd-a552-47eb6d3e6a84.mov

Could you try again with only the attributes values that you need ?
Thanks!

**Comment by guestisp** [comment]
Here it doesn't work:

![image](https://user-images.githubusercontent.com/3944655/215447419-b2f3ae56-2f0a-4a44-9eb1-6f4f89d99d55.png)

[Video del 30-01-2023 11:06:55.webm](https://user-images.githubusercontent.com/3944655/215447669-999e9049-a135-4305-bb80-ecb1bd85ae2c.webm)

**Comment by sallemiines** [comment]
Hello @guestisp 

Following your steps , unfortunately I didn't manage to reproduce your issue , could you please provide us more infos / screenshots ? 

Thank you !

**Comment by guestisp** [comment]
i don't know what to post more, as the video show that the options are not selectable  and there is also an admin screenshot of the current combinaton

can you please post your exact test configuration so that i can try to reproduce on my side ?

**Comment by guestisp** [comment]
If you look carefully, the "Nessuno" option doesn't have any "number of supports" set other than "none"
The other supports (ventosa, stadard magnet, ..." have all options set except "none"

As the first selected option is "No support" + "None", probably prestashop is not able to re-add the missing options in the "number of support" select when selection something different than "No support"

Try this:
- delete all combinations from BO
- create combination for "Color: Orange, Green", "Support type: None, Suction, Magnet", "Number of supports": None, 1, 2
- from the combination page, for each color, DELETE any "Number of supports" related to "Support Type: NONE" EXCEPT "None"
- from the combination page, for each color, DELETE "Support Type: None" from any combination related to "Number of supports: 1 or 2"

in other words, the "Support Type: None" option should only have "Number of supports: None" set
the "Support Type != None" options should onyl have "Number of supports != None" set.

Valid combination : ORANGE + Support Type: None + Number of Supports: None
Valid combination : ORANGE + Support Type: Magnet + Number Of Supports: 1, 2, ....

in FO, as the starting combination is "Orange + None + None", when you swap to "Magnets", prestashop it's unable to keep the "Number of supports" NONE selected because it's not in the option list and doesn't do anything. It should select the first option available (in this case, "1")

Hope it's clear, if not, I can provide (privately) access to a test e-commerce.

**Comment by florine2623** [comment]
Hello again @guestisp ,

Despite our efforts, we were not able to reproduce your issue with the provided information. It seems that your issue is not a PrestaShop core bug but most likely a server configuration or customization problem.

We use GitHub issues only to discuss about bugs and new features in the PrestaShop project. If you have questions about using PrestaShop or if you need help with your shop, please consider one of our [support plans](https://www.prestashop.com/support).

Alternatively, you can also ask for help in the [community forums](https://www.prestashop.com/forums/) or in the [public Slack channel](https://github.com/PrestaShop/open-source/tree/master/slack).

We will close this issue.

Thank you

**Comment by guestisp** [comment]
it's a fresh install, can't be a server bug, as it's a JavaScript issue.

it's strange that you aren't able to reproduce the issue, probably you configured the combination differently than me.

can i see you current combination page?

**Comment by guestisp** [comment]
So ? No other news ? Still doesn't work even on a fresh install.
Could you please show your exact combination configuration to double check with mine ?
